mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/model/impl/MessageParser.java

```diff
@@ -208,7 +206,7 @@ private boolean isAttachment(Entity part, Context context) {
             .map(ATTACHMENT_CONTENT_TYPES::contains)
             .orElse(false);
 
-        if (bodyTextCriterion) {
+        if (context == Context.BODY && isTextPart(part)) {
```

Comment:
this should go first, no?

Reply:
I wanted to have decisions grouped in the same part of the method to ease reading...